File: slnSopimusrekisteriLuokat/Sopimusrekisteri.BLL_CF/Aktiviteetti.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Sopimusrekisteri.BLL_CF
{
  public class Aktiviteetti : IEditable
  {
		#region Propertyt

		public int Id {get; private set;}
		public int? TahoId {get; set;}
		public int? TSopimusId {get; set;}
		public int? YhteystapaId {get; set;}
		public DateTime? Paivamaara {get; set;}
		public string Kuvaus {get; set;}
		public int? AktiviteetinLajiId {get; set;}
		public DateTime? SeuraavaYhteyspaiva {get; set;}
		public int? StatusId {get; set;}
		public string LiitetiedostoPolku {get; set;}
		public DateTime Luotu {get; set;}
		public string Luoja {get; set;}
		public DateTime? Paivitetty {get; set;}
		public string Paivittaja {get; set;}
		

		#endregion

		#region Entiteettiviittaukset

		public virtual Taho Taho {get; set;}
		public virtual Sopimus TSopimus {get; set;}
		public virtual AktiviteettiYhteystapa Yhteystapa {get; set;}
		public virtual AktiviteetinLaji AktiviteetinLaji {get; set;}
		public virtual AktiviteetinStatus Status {get; set;}
		//public virtual aspnet_Users Kontaktoija {get; set;}

		#endregion

		#region Metodit

		public override bool Equals(object obj)
		{
			var tmp = obj as Aktiviteetti;
			if (tmp == null) return false;
			return tmp.Id == Id;
		}

		public override int GetHashCode()
		{
			return Id.GetHashCode();
		}

		public override string ToString()
		{
			return Id.ToString();
		}

		#endregion
  }
}

File: slnSopimusrekisteri/appSopimusrekisteri.DTO/Aktiviteetti.vb
Public Class Aktiviteetti
    Public Property Luoja As String
    Public Property Luotu As DateTime?
    Public Property Paivittaja As String
    Public Property Paivitetty As DateTime?

    Public Property Id As Integer
    Public Property TahoId As Integer?
    Public Property Taho As Taho
    Public Property SopimusId As Integer?
    Public Property Sopimus As Sopimus
    Public Property KontaktoijaGuid As Guid?

    Public Property Paivamaara As DateTime?
    Public Property SeuraavaYhteydenottoPaivamaara As DateTime?
    Public Property Kuvaus As String
    Public Property Liitetiedostopolku As String

    Public Property YhteydenottotapaId As Integer?
    Public Property Yhteydenottotapa As String

    Public Property LajiId As Integer?
    Public Property Laji As String

    Public Property StatusId As Integer?
    Public Property Status As String
End Class
